src/ApiProvider.php

```diff
@@ -80,4 +92,14 @@ public function createClient(): ClientInterface
 
         return $client;
     }
+
+    /**
+     * Sets an optional rate limiting function.
+     *
+     * @param \Closure
+     */
+    public function setRateLimiter(\Closure $rateLimiter)
```

Comment:
As the previous comment, I would use `callable` typehint or no typehinting at all and add `is_callable` check in method body (with `InvalidArgumentException` throw if check fails).